src/jit/codegencommon.cpp

```diff
@@ -10677,7 +10671,46 @@ void CodeGen::genRestoreCalleeSavedFltRegs(unsigned lclFrameSize)
             offset -= XMM_REGSIZE_BYTES;
         }
     }
+    genVzeroupperIfNeeded();
 }
+
+// Generate Vzeroupper instruction as needed to zero out upper 128b-bit of all YMM registers so that the 
+// AVX/Legacy SSE transition penalties can be avoided
+//
+// Params
+//   is256bitAVX - Flag to check if the function contains 256-bit AVX instruction and generate Vzeroupper
+//      instruction, otherwise check if the function contains AVX instruciton (either 128-bit or 256-bit). 
```

Comment:
This parameter name is a little confusing. It might be better to call this `check256bitOnly` to make it clear that this is called when we only need the vzeroupper if we've used 256-bit AVX.